### ISSUE TITLE
Vite: SSR compile error with importing *.client.ts

### ISSUE THREAD

**Issue by na2hiro** [remix-run/remix#7972]
### What version of Remix are you using?

2.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

When a route imports *.client.ts, it results in compile error

```tsx
import { myAlert } from '~/something.client';

export default function Index() {
  return (
...
      <button onClick={() => myAlert('clicked')}>
        click to invoke client-only method
      </button>
  )
}
```

Here's repro https://stackblitz.com/edit/github-suecku?file=app/something.client.ts,app/routes/_index.tsx

### Expected Behavior

`vite build --ssr` succeeds, and `myAlert` is provided `undefined` in server (according to https://remix.run/docs/en/main/file-conventions/-client)

### Actual Behavior

`vite build --ssr` results in compile error
```
vite v4.5.0 building SSR bundle for production...

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

✓ 7 modules transformed.
✓ built in 87ms
"myAlert" is not exported by "app/something.client.ts", imported by "app/routes/_index.tsx".
file: /home/projects/github-suecku/app/routes/_index.tsx:2:9
1: import type { MetaFunction } from '@remix-run/node';
2: import { myAlert } from '~/something.client';
            ^
3: 
4: export const meta: MetaFunction = () => {
error during build:
RollupError: "myAlert" is not exported by "app/something.client.ts", imported by "app/routes/_index.tsx".
    at error (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:2312:30)
    at Module.error (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:13770:16)
    at Module.traceVariable (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:14200:29)
    at ModuleScope.findVariable (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:12640:39)
    at FunctionScope.findVariable (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:7149:38)
    at ChildScope.findVariable (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:7149:38)
    at ReturnValueScope.findVariable (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:7149:38)
    at Identifier.bind (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:8344:40)
    at CallExpression.bind (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:5917:23)
    at CallExpression.bind (/home/projects/github-suecku/node_modules/rollup/dist/es/shared/node-entry.js:9915:15)
```

**Comment by cliffordfajardo** [comment]
Also running into this during a migration

**Comment by pcattori** [comment]
Fixed by #8200
